File: javascript/space-age/space-age.js
var SpaceAge = function(seconds) {
  this.seconds = seconds;

  var planetYear = function(planet_delta) {
    var earth = seconds / 31557600.0;
    var planet = earth / planet_delta;
    return Math.round( planet * 100 ) / 100.0;
  };

  this.onEarth = function() {
    return planetYear(1);
  };

  this.onMercury = function() {
    return planetYear(0.2408467);
  };

  this.onVenus = function() {
    return planetYear(0.61519726);
  };

  this.onMars = function() {
    return planetYear(1.8808158)
  };

  this.onJupiter = function() {
    return planetYear(11.862615);
  };

  this.onSaturn = function() {
    return planetYear(29.447498);
  };

  this.onUranus = function() {
    return planetYear(84.016846);
  };

  this.onNeptune = function() {
    return planetYear(164.79132);
  };
}

module.exports = SpaceAge;

File: java/space-age/src/main/java/SpaceAge.java
public class SpaceAge {
  double seconds = 0;

  SpaceAge(double seconds) {
    this.seconds = seconds;
  }

  public double getSeconds() {
    return this.seconds;
  }

  public double onEarth() {
    return this.getSeconds() / 31557600;
  }

  public double onMercury() {
    return this.onEarth() / 0.2408467;
  }

  public double onJupiter() {
    return this.onEarth() / 11.862615;
  }

  public double onSaturn() {
    return this.onEarth() / 29.447498;
  }

  public double onMars() {
    return this.onEarth() / 1.8808158;
  }

  public double onUranus() {
    return this.onEarth() / 84.016846;
  }

  public double onNeptune() {
    return this.onEarth() / 164.79132;
  }

  public double onVenus() {
    return this.onEarth() / 0.61519726;
  }

}
